docs/native-modules.md

```diff
@@ -293,17 +293,25 @@ Then we define constants, and it's as easy as creating a public field and giving
 
 It's just as easy to add custom methods, by attributing a public method with `REACT_METHOD`. In FancyMath we have one method, `add`, which takes two doubles and returns their sum. Again, we've specified the optional `name` argument in the `REACT_METHOD` macro-attribute so in JS we call `add` instead of `Add`.
 
-Methods that return more complex types (like `std::string`) can be implemented by returning void and accepting a bool and a `ReactPromise<JSValue>`:
+Native modules do not have the ability to check that the parameter types and the number of parameters match between what's called from JavaScript and what the native code accepts. However, the framework will validate that the number of promises-like parameters match: if the JavaScript API is async, it will expect that there is one "promise-like" parameter in the native method implementation signature.

```

Comment:
>Native [](start = 0, length = 6)

I am not sure that it is correct to use the term 'promise-like'. It would be better to say that an asynchronous method has different way to return result. It can be one callback, two callbacks, or a promise.

Reply:
@vmoroz We should:
- Define what a callback is. Is it strictly a `void (*) ()`? a `std::function`?  is it anything with an `operator()` ?
- What types they take/return
- What the difference between 1 or 2 is
- When you would use a callback, two callbacks or a promise

Reply:
Filed 283